src/main/java/me/drepic/proton/ProtonManager.java

```diff
@@ -204,20 +206,22 @@ private void registerMessageHandler(Plugin plugin, Object object){
                 }
 
                 if(!this.contextClassMap.containsKey(context)){
-                    this.contextClassMap.put(context, parameterClass);
-                    List<BiConsumer<Object, MessageAttributes>> biConsumers = new ArrayList<>();
-                    biConsumers.add(wrappedBiConsumer);
-                    this.messageHandlers.put(context, biConsumers);
-                    try {
-                        registerMessageContext(context);
-                    } catch (IOException e) {
-                        e.printStackTrace();
+                    synchronized(registrationLock){
```

Comment:
According to https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/ArrayListMultimap.html:

> This class is not threadsafe when any concurrent operations update the multimap. Concurrent read operations will work correctly. To allow concurrent update operations, wrap your multimap with a call to Multimaps.synchronizedListMultimap(com.google.common.collect.ListMultimap<K, V>). 

That's probably preferred